clash-lib/src/Clash/Netlist/Types.hs

```diff
@@ -586,14 +594,58 @@ data Seq
       [(Maybe Literal,[Seq])]  -- ^ List of: (Maybe match, RHS of Alternative)
   deriving Show
 
+data Blocking
+  = NonBlocking
+  | Blocking
+  deriving (Binary, Eq, Generic, Hashable, NFData, Show)
+
+instance Semigroup Blocking where
+  NonBlocking <> y = y
+  Blocking    <> _ = Blocking
+
+data Usage
+  = Cont
+  -- | Used for non-blocking procedural assginment
+  | Proc Blocking
+  deriving (Binary, Eq, Generic, Hashable, NFData, Show)
+
+instance Semigroup Usage where
+  Cont    <> y      = y
+  Proc x  <> Proc y = Proc (x <> y)
+  Proc x  <> _      = Proc x
+
+instance FromJSON Usage where
+  parseJSON = Aeson.withText "Usage" $ \case
+    "continuous"  -> pure Cont
+    "nonblocking" -> pure (Proc NonBlocking)
+    "blocking"    -> pure (Proc Blocking)
+    str           -> fail ("Could not parse usage: " <> show str)
```

Comment:
Explain valid options in error message.
Probably use uppercased values, to blend in better with other blackbox fields.

Reply:
Now it reads:

```
Failed to decode YAML:

                  /.../prims/vhdl/Clash_Sized_Vector.primitives.yaml

                Decoder reported:

                  Aeson exception:
                  Error in $[15].outputUsage: Could not parse usage: "nonblocking"
                  Recognized values are 'Continuous', 'NonBlocking' and 'Blocking'
```